src/linker/Linker/LinkContext.cs

```diff
@@ -513,6 +518,9 @@ public void LogWarning (string text, int code, MessageOrigin origin, string subc
 				return;
 
 			var warning = MessageContainer.CreateWarningMessage (this, text, code, origin, subcategory);
+			if (OutputWarningSuppressions && warning != MessageContainer.Empty && origin.MemberDefinition != null)
+				WarningSuppressionWriter.AddWarning ((code, origin.MemberDefinition));
```

Comment:
This should be in the `LogMessage` if somebody calls `CreateWarningMessage` manually and then calls `LogMessage` that warning would be missed.